src/main/scala/EnsimeExtrasPlugin.scala

```diff
@@ -221,4 +230,38 @@ object EnsimeExtrasPlugin extends AutoPlugin {
     extracted.append(newSettings, state)
   }
 
+  def formatOnlyTask: Def.Initialize[InputTask[Unit]] = InputTask(
+    (s: State) => Def.spaceDelimited()
+  ) { (argTask: TaskKey[Seq[String]]) =>
+      (argTask, sourceDirectories, scalariformPreferences, scalaVersion, streams).map {
+        (files, dirs, preferences, version, s) =>
+          if (files.isEmpty) throw new IllegalArgumentException("needs a file")
+          files.foreach(arg => {
+            val input: File = file(arg).getCanonicalFile
+            val sourceDirs = dirs.map(_.getCanonicalFile)
+
+            val here = sourceDirs.exists { dir => input.getPath.startsWith(dir.getPath) }
+            if (!here || !input.exists())
+              throw new IllegalArgumentException(s"$arg not associated to $sourceDirs")
+
+            if (!input.getName.endsWith(".scala"))
+              throw new IllegalArgumentException(s"only .scala files are supported: $arg")
+
+            try {
+              val contents = IO.read(input)
+              val formatted = ScalaFormatter.format(
+                contents,
+                preferences,
+                scalaVersion = version split "-" head
```

Comment:
ok cool, then there is precedent. It's probably something like CrossVersion.partialVersion(scalaVersion)